General cleanup and package updates

opsyne opened the pull request:
A lot of the node packages were out of date, so I upgraded them and updated the gulpfile for gulp 4. I also added build information to the README, removed execute bits on some files that didn't need it, added a Gemfile for jekyll, and updated the jekyll config.

opsyne committed:
Remove unnecessary execute bits

opsyne committed:
Create Gemfile and Gemfile.lock, add jekyll directories to gitignore

opsyne committed:
Upgrade node packages, rewrite gulpfile for gulp 4

Gulp 4 changed the interface for specifying task dependencies.

opsyne committed:
Update excludes, add the jekyll-paginate plugin

Pagination was specified in the config, but the plugin wasn't added.

opsyne committed:
Add build instructions and license info to README

opsyne committed:
add scripts to package.json

morganava commented:
o/ if you could update this with the latest web nonsense from the past 8 months we should be able to get his merged (thanks btw ;) )

morganava commented:
ping @chrisculnane

opsyne committed:
forcibly update glob-parent to fix vulnerabilities

opsyne committed:
fix dev script

opsyne committed:
Update ruby packages

opsyne commented:
I had to break a few js dependencies in package-lock to fix the security warnings from npm, but build and dev still work fine so everything should be alright.

chrisculnane commented on Gemfile:
```diff
@@ -0,0 +1,14 @@
+
+# gem "rails"
+
+gem "jekyll", "~> 4.2"
```
GitHub Pages currently supports Jekyll 3.9.2: https://pages.github.com/versions/ I don't think it will make much difference in this case because the remaining dependencies are supported and it isn't a particularly complicated site. However, I'm a little uneasy using a major version different from GitHub Pages, although I don't think it will break during the build on GitHub pages. 

I think there are three options to resolve it:

1. Ignore the dependency difference, merge and see if it builds correctly, we can always rollback if it doesn't
2. Use the GitHub pages gem which uses a compatible version of Jekyll (added as a suggestion, but ignore if we aren't going for this option). The downside is that it may have out of date dependencies, nullifying the point of the updates. 
3. Configure a dedicated GitHub Action workflow to build outside the default GitHub pages workflow and use the specified version of Jekyll

Suggestions/comments welcome on the preferable option. I'm tempted to try 1 with the fallback of if the build fails rolling back and then choosing between option 2 or 3. 

@pospeselr, @opsyne any thoughts/preferences?

```suggestion
gem "github-pages", "~> 227", group: :jekyll_plugins
```

opsyne replied:
i think using the "github-pages" gem is best. it seems like the easiest way to get dev builds and github-pages builds playing nicely

chrisculnane replied:
Great, I'll commit the suggestion and merge. Thanks!

chrisculnane committed:
Switch to github-pages gem

Switch to github-pages gem to maintain compatibility between local and GitHub pages Jekyll versions.
